feat: 서랍장 메인 화면 구현

Hanna922 opened the pull request:
## 1️⃣ 어떤 작업을 했나요? (Summary)

- resolved #21 

### 기존 코드에 영향을 미치지 않는 변경사항
- drawer에서 '더보기' 버튼 역할을 하는 GrayButton을 추가했습니다.
- drawer main에서 사용되는 image를 4x로 drawer/assets/ 경로에 추가했습니다.
- main page 화면을 구현하였습니다.

### 기존 코드에 영향을 미치는 변경사항
- 첫 번째 태스크였던 Button, Footer의 css 단위를 px -> rem 으로 수정했습니다.
- Button에 cursor, hover style을 추가했습니다.
  - hover color 디자이너님께 전달 완료
- @2wndrhs Header 스타일을 살짝 수정했습니다.
  - 스크롤과 관계없이 항상 보이도록 설정 & background color 설정
- @seocylucky Category 스타일을 살짝 수정했습니다.
  - 피그마 dev mode 기반 rem 값으로 수정 & 스크롤과 관계없이 항상 보이도록 설정
- drawer pages export 형식을 export const로 변경했습니다.
- 이제는 필요가 없을 것 같아 src/components/Test 폴더를 삭제합니다.
- Footer style.ts의 text -> span으로 변경했습니다.

### 구현 화면
![image](https://github.com/yourssu/Soomsil-Web/assets/84809236/99f8825c-576b-49ff-8e1c-57ea0a7b2463)

![image](https://github.com/yourssu/Soomsil-Web/assets/84809236/b02ac8f4-827c-4a82-876d-25202769c949)

## 2️⃣ 알아두시면 좋아요!
1. 피그마에선 Footer도 스크롤과 관계없이 항상 하단에 배치되는 상태인데용
뭔가... 그렇게 하였더니 살짝... 어색한 것 같아서,,, 요거 회의 때 질문 드리면 좋을 것 같다고 생각했슴다...!
2. 현재 Register page의 Card 배치 map key가 index로 설정되어있는 상태인데용
요건 나중에 데이터 연결 시 card id로 수정이 필요할 것 같아 주석 적어두었습니당

## 4️⃣ 체크리스트 (Checklist)

- [x] `main` 브랜치의 최신 코드를 `pull` 받았나요?

nijuy commented on src/drawer/assets/drawer_main_image1.png:
사진 속 contents 보니까 임시 이미지 같은데 이거 최종 이미지 언제 나오는지도 물어봐야겠어요
이미지는 전혀 급하진 않지만 궁금하니께 🥸

Hanna922 replied:
엥 이거 임시 이미지였군요 다음 회의 때 바로 여쭤봅시다 ㅎㅅㅎ

nijuy commented on src/drawer/components/Footer/Footer.style.ts:
```diff
@@ -7,12 +7,12 @@ export const StyledFooter = styled.footer`
   background-color: ${({ theme }) => theme.color.bgPressed};
 `;
 
-export const StyledFooterLogo = styled.text``;
```
이 부분을 왜 교체하신 건지 궁금했는데, `text`를 사용했을 때 콘솔에 워닝이 뜨고 있었군요.,.,........

![image](https://github.com/yourssu/Soomsil-Web/assets/87255462/5f541407-3022-44a9-96e3-2230fe36ff66)

알아보니 `text` 태그는 `svg` 내부에서만 유효한 태그고, 외부에서는 인식을 못해서 저런 경고가 🙄 하나 알아갑니덩

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text

nijuy commented on src/drawer/components/GrayButton/GrayButton.style.ts:
```diff
@@ -0,0 +1,19 @@
+  border: none;
+  background-color: ${({ theme }) => theme.color.buttonBG};
+  color: ${({ theme }) => theme.color.textSecondary};
+  font-style: ${({ theme }) => theme.typo.caption0};
```
저번에 쭌도 라인 10처럼 typo를 적용했다가 font-size 적용이 안되는 문제가 있었는데요!

콘솔 열어보니 현재도 동일한 문제가 발생하고 있어 수정이 필요할 거 같습니다

![image](https://github.com/yourssu/Soomsil-Web/assets/87255462/b6a25fc0-fa0f-4aec-ad37-210beb01d729)

Hanna922 replied:
어머.......... YDS 코드 짠 사람이 font-style에다가 저걸 집어넣었네요 머쓱 ^^^... 수정하겠슴당~!

nijuy commented on src/drawer/components/GrayButton/GrayButton.tsx:
현재 drawer/components 하위에 `Button` `GrayButton` 폴더가 동시 존재하는 상황인데,
용도를 드러내는 유의미한 이름으로 교체하면 더 좋을 거 같아요 🤔
그치만. 네이밍이 항상 젤 어려운 문제이므로............. 저도 몬가 좋은 이름이 없을지 고민해볼게요

nijuy commented on src/drawer/pages/Ranking/Ranking.style.ts:
```diff
@@ -0,0 +1,52 @@
+
+export const StyledCardContainer = styled.section`
+  display: grid;
+  grid-template-columns: 25.5rem 25.5rem 25.5rem;
```
이렇게 같은 크기로 n등분 할 경우 `repeat`를 쓰면 더 깔끔한 코드가 될 거 같아요!!!!!

```suggestion
  grid-template-columns: repeat(3, 25.5rem);
```

nijuy reviewed (commented):
고생하셨어요!! 우왕. 이제 점점......... 그럴싸해지네요

다만 현재 레이아웃이 많이 깨지는 상태라 수정이 쪼끔 필요할 거 같아요!!
제 노트북 화면이 쫌 작은 편인데, 아래처럼 보이고 있읍니다

|100%|100% 미만 배율|
|---|---|
|![image](https://github.com/yourssu/Soomsil-Web/assets/87255462/3f87d636-1fc2-441e-a8a6-dea735db19d1)|![image](https://github.com/yourssu/Soomsil-Web/assets/87255462/cac8c9b4-3af2-40dd-921f-47f98dcc6bff)|

Hanna922 commented:
> 다만 현재 레이아웃이 많이 깨지는 상태라 수정이 쪼끔 필요할 거 같아요!! 제 노트북 화면이 쫌 작은 편인데, 아래처럼 보이고 있읍니다

100%가 보리 원래 노트북 화면이신거죵?? 헤더도 오른쪽 여백이 크게 보이네용 그때 준이 언급하셨던 문제 같은데 흠흠 어떻게 할지 좀 고민이 됩니당

nijuy commented:
> @nijuy 100%가 보리 원래 노트북 화면이신거죵?? 헤더도 오른쪽 여백이 크게 보이네용 그때 준이 언급하셨던 문제 같은데 흠흠 어떻게 할지 좀 고민이 됩니당

마자요!!!
컴포넌트끼리 겹치는 이유가 카테고리를 fixed로 고정해뒀기 때문인(거 같은)데,
fixed를 사용하지 않는 방법을 사용하는 게 좋을 거 같아요

[링크](https://www.inflearn.com/courses) 들어가서 쭉 확대해보시면 일정 조건부터 강의 분류가 사라지는 걸 보실 수 있는데, 이런 방향으로 가도 좋을 거 같습니다~ 근데 이제....... 이건 디자이너한테 말은 해야할 거 같은..............

Hanna922 commented:
> 넹 제 원래 노트북 화면 마자요!!! 컴포넌트끼리 겹치는 이유가 카테고리를 fixed로 고정해뒀기 때문인(거 같은)데, fixed를 사용하지 않는 방법을 사용하는 게 좋을 거 같아요
> 
> [링크](https://www.inflearn.com/courses) 들어가서 쭉 확대해보시면 일정 조건부터 강의 분류가 사라지는 걸 보실 수 있는데, 이런 방향으로 가도 좋을 거 같습니다~ 근데 이제....... 이건 디자이너한테 말은 해야할 거 같은..............

으으음 카테고리 분류를 없애는 건 가능한데 그럼 14인치 노트북 이하에선 어디서 카테고리를 보여줄지 디자이너님들께 여쭤봐야 할 것 같아요
fixed를 사용하지 않고 고정시키는 것도 가능한데 첨부해주신 노트북 화면 상에선 이미 카드 컴포넌트 크기가 노트북 전체를 차지하고 있어서 카드 컴포넌트의 크기도 작아져야 할 것 같슴다........
회의 때 1920px보다 작은 화면 디자인을 논의해봐야 할 것 같은데,, 이게 쪼오끔 시간이 걸릴 것 같아서,,,, 해당 PR에선 카테고리 fixed만 풀어둬도 괜찮을까용?

JjungminLee commented on src/drawer/components/GrayButton/GrayButton.style.ts:
```diff
@@ -0,0 +1,19 @@
+  color: ${({ theme }) => theme.color.textSecondary};
+  ${({ theme }) => theme.typo.caption0};
+  cursor: pointer;
+  &:hover {
```
수고하셨습니당!! 제가 다음 작업으로 reset.css작업이랑 global.css작업을 할 예정인데 button에 필요한 속성들 (`border: none`으로 만들기,`cursor: pointer`로두기 등등)을 추가해둘게요!  다른 분들 css코드 보면서 느끼는게 css reset코드가 없어서 작업하실때 번거로우실것 같아용 ㅜㅜ

nijuy replied:
당신멋져당신최고야
